xtask/codegen/gritql.ungram

```diff
@@ -0,0 +1,554 @@
+
+GritPatternLimit =
+    pattern: AnyGritPattern 'limit' limit: GritIntLiteral
+
+// statement, in the engine this is a predicate that always evaluates to true
+// This is useful for initializing variables at the root of a pattern definition
+GritAssignmentAsPattern =
+    container: AnyGritContainer '=' pattern: AnyGritPattern
+
+// Primarily used for initializing variables at the root of a pattern definition
+GritPatternAccumulate =
+    left: AnyGritPattern '+=' right: AnyGritPattern
+
+GritPatternWhere =
+    pattern: AnyGritPattern 'where' side_condition: AnyGritPredicate
+
+AnyGritLiteral =
+    GritCodeSnippet
+    | GritStringLiteral
+    | GritIntLiteral
+    | GritDoubleLiteral
+    | GritBooleanLiteral
+    | GritUndefinedLiteral
+    | GritMap
+    | GritList
+    | GritBogusLiteral
+
+GritPatternNot =
+    GritNot
+    pattern: AnyGritPattern
+
+GritPatternOr = 'or' '{' patterns: GritPatternList '}'
+
+GritPatternOrElse = 'orelse' '{' patterns: GritPatternList? '}'
+
+GritPatternAny = 'any' '{' patterns: GritPatternList? '}'
+
+GritPatternAnd = 'and' '{' patterns: GritPatternList? '}'
+
+GritPatternMaybe = 'maybe' pattern: MaybeCurlyGritPattern
+
+GritPatternAfter = 'after' pattern: AnyGritPattern
+
+GritPatternBefore = 'before' pattern: AnyGritPattern
+
+GritPatternContains =
+    'contains'
+    contains: MaybeCurlyGritPattern
+    GritPatternContainsUntilClause?
+
+GritPatternContainsUntilClause =
+    'until'
+    until: AnyGritPattern
+
+GritPatternIncludes = 'includes' includes: MaybeCurlyGritPattern
+
+GritRewrite =
+    left: AnyGritPattern
+    annotation: GritAnnotation?
+    '=>'
+    right: AnyGritPattern
+
+// --- conditional pattern ------------
+GritPatternIfElse =
+    'if'
+    '('
+    if_predicate: AnyGritPredicate
+    ')'
+    then_pattern: MaybeCurlyGritPattern
+    GritPatternElseClause?
+
+GritPatternElseClause =
+    'else'
+    else_pattern: MaybeCurlyGritPattern
+// --- conditional pattern ------------
+
+GritWithin =
+    'within'
+    pattern: MaybeCurlyGritPattern
+
+GritBubbleScope = '(' GritVariableList? ')'
+
+GritBubble =
+    'bubble'
+    variables: GritBubbleScope?
+    pattern: MaybeCurlyGritPattern
+
+MaybeGritNamedArg =
+    AnyGritPattern
+    | GritNamedArg
+    | GritBogusNamedArg
+
+GritNamedArg =
+    name: GritName
+    '='
+    pattern: AnyGritPattern
+
+GritNamedArgList = MaybeGritNamedArg (',' MaybeGritNamedArg)* ','?
+
+// maybe we should have a different fieldname for each choice?
+GritNodeLike =
+    name: GritName
+    '('
+    named_args: GritNamedArgList?
+    ')'
+
+GritLike =
+    'like'
+    GritLikeThreshold?
+    '{'
+    example: AnyGritPattern
+    '}'
+
+GritLikeThreshold =
+    '('
+    threshold: AnyGritPattern
+    ')'
+
+GritMap =
+    '{'
+    elements: GritMapElementList?
+    '}'
+
+GritMapElementList = GritMapElement (',' GritMapElement)* ','?
+
+GritMapElement =
+    key: GritName
+    ':'
+    value: AnyGritPattern
```

Comment:
`pattern`?

Reply:
On second thought, no. They are really referred to as values: https://docs.grit.io/language/patterns#maps